### ISSUE TITLE
Redshift: doesn't parse encoding in add column statement

### ISSUE THREAD

**Issue by ivan-ishchanka-c** [sqlfluff/sqlfluff#6947]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

SQLFluff Redshift dialect doesn't support the usage of `ENCODE encoding` in `ADD COLUMN` statement. SQLFluff fails to parse  `ALTER TABLE` statements if you try to add a column with some `ENCODING`

### Expected Behaviour

I expected SQLFluff to successfully parse the SQL file, including the `ENCODE encoding` option inside a `ADD COLUMN` statement, because redshift supports it: https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_TABLE.html

### Observed Behaviour

SQLFluff cannot parse the file when encoding is used in `ADD COLUMN` statement. The error message is:

```
L:   2 | P:  29 |  PRS | Line 2, Position 29: Found unparsable section: 'ENCODE
                       | ZSTD;'
```

This seems to be due to missing parsing support for `ENCODE encoding` in `ADD COLUMN` statement in redshift dialect

### How to reproduce

**SQL file**:
file.sql
```SQL
ALTER TABLE my_schema.my_table
ADD COLUMN my_column BIGINT ENCODE ZSTD;
```

**command**:

```bash
sqlfluff lint file.sql --dialect redshift
```

**result**:
```
== [file.sql] FAIL                                                                                                                                                                                                                                                                     
L:   2 | P:  29 |  PRS | Line 2, Position 29: Found unparsable section: 'ENCODE
                       | ZSTD;'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
```

**Observation**:
Removing the `ENCODE ZSTD` statement does not return the error and sqlfluff parses the file with no issue 

### Dialect

redshift

### Version

sqlfluff, version 3.4.0
Python 3.13.5

### Configuration

[sqlfluff]
dialect = redshift

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
